Add the Projects section

HidemichiShimura opened the pull request:
## Proposed Changes

#### Code changes

- Created `<ProjectDescription />`, `<ProjectThumbnail />`, `<Projects />`
- Embedded `<Projects />` in `<Main />`
- Added the created components in the entry point
- Added constants to use in the Projects section
- Fixed the common component `<LinkButton />` by changing the breakpoint and adding the `target` property

#### Issues affected

- Resolves #40
- Partially addresses #41

## Additional Info

- none

## Screenshots and/or video

- Mobile view
![image](https://user-images.githubusercontent.com/110521018/215689442-b3d096bf-26e0-4034-a46e-45b968507551.png)

- Tablet / Laptop view
![image](https://user-images.githubusercontent.com/110521018/215689562-3c71a01c-56cc-434b-9b90-8061316672ab.png) 

## Testing Plan

- Check if there is no UI bug and if there is no error in the console with Google developer tools

## Checklist

#### Basics

- [x] Local QA is complete
- [x] No debugging `console.log()` statements
- [x] Commented-out code removed
- [x] Unused code removed (imports, functions, styles, etc - if it's not used anywhere, it's not in this PR)

yuya-minamide commented:
I noticed that only this photo is crushed
If it bothers you, you may want to modify it.
<img width="413" alt="michi-project" src="https://user-images.githubusercontent.com/110059708/215705959-95282b6e-8483-4dc3-bb35-229b2e66a2c6.png">

At mobile size, it might be easier to see if the space between the GitHub button and the next project's photo were a little wider.
<img width="317" alt="michi-project-smartphone" src="https://user-images.githubusercontent.com/110059708/215707329-f5046572-6ed3-4bcd-9249-ad9262c1c774.png">

HidemichiShimura commented:
> I noticed that only this photo is crushed
If it bothers you, you may want to modify it.

Thank you for noticing that.
I replaced the image with another one that has no crush.

> At mobile size, it might be easier to see if the space between the GitHub button and the next project's photo were a little wider.

I see that.
I added more space between projects not only for mobile but for the other devices because the Tablet / Laptop view looked better with more space

HidemichiShimura commented on src/components/main/projects/ProjectDescription.jsx:
```diff
@@ -0,0 +1,48 @@
+`;
+
+const ProjectTitle = styled.h3`
+  font-size: calc(16px + 1.5vw);
```
There is no reason for using it.
I will create another PR to standardize font sizes used in the whole page because it seems weird that calc() is applied only to the title as you say.

yuya-minamide reviewed (approved):
LGTM
